[Flaky Test] TestConfigAcceptValid/Kerberos_with_keytab

## Flaky Test

* **Test Name:** TestConfigAcceptValid
* **Link:** https://github.com/elastic/beats/blob/62f297169fafaf458ccd2963ca6ea51f1832e5d1/libbeat/outputs/kafka/config_test.go#L38-L46
* **Branch:** master
* **Artifact Link:** If available, attach the generated zip artifact associated with the stack trace for this failure.
* **Notes:** It's failing on travis-ci when running `make testsuite`

### Stack Trace

```
github.com/elastic/beats/v7/libbeat/outputs/kafka]: exit status 1
--- FAIL: TestConfigAcceptValid (0.00s)
    --- FAIL: TestConfigAcceptValid/Kerberos_with_keytab (0.00s)
        config_test.go:69: Failure creating sarama config: kafka: invalid configuration (Net.SASL.GSSAPI.Username must not be empty when GSS-API mechanism is used)
FAIL
```

--- a/auditbeat/auditbeat.reference.yml
+++ b/auditbeat/auditbeat.reference.yml
@@ -796,7 +796,7 @@ output.elasticsearch:
   # The service principal name.
   #kerberos.service_name: HTTP/my-service@realm
 
-  # Name of the Kerberos user. It is used when auth_type is set to password.
+  # Name of the Kerberos user.
   #kerberos.username: elastic
 
   # Password of the Kerberos user. It is used when auth_type is set to password.

--- a/filebeat/filebeat.reference.yml
+++ b/filebeat/filebeat.reference.yml
@@ -1502,7 +1502,7 @@ output.elasticsearch:
   # The service principal name.
   #kerberos.service_name: HTTP/my-service@realm
 
-  # Name of the Kerberos user. It is used when auth_type is set to password.
+  # Name of the Kerberos user.
   #kerberos.username: elastic
 
   # Password of the Kerberos user. It is used when auth_type is set to password.

--- a/heartbeat/heartbeat.reference.yml
+++ b/heartbeat/heartbeat.reference.yml
@@ -940,7 +940,7 @@ output.elasticsearch:
   # The service principal name.
   #kerberos.service_name: HTTP/my-service@realm
 
-  # Name of the Kerberos user. It is used when auth_type is set to password.
+  # Name of the Kerberos user.
   #kerberos.username: elastic
 
   # Password of the Kerberos user. It is used when auth_type is set to password.

--- a/journalbeat/journalbeat.reference.yml
+++ b/journalbeat/journalbeat.reference.yml
@@ -734,7 +734,7 @@ output.elasticsearch:
   # The service principal name.
   #kerberos.service_name: HTTP/my-service@realm
 
-  # Name of the Kerberos user. It is used when auth_type is set to password.
+  # Name of the Kerberos user.
   #kerberos.username: elastic
 
   # Password of the Kerberos user. It is used when auth_type is set to password.

--- a/libbeat/_meta/config.reference.yml.tmpl
+++ b/libbeat/_meta/config.reference.yml.tmpl
@@ -677,7 +677,7 @@ output.elasticsearch:
   # The service principal name.
   #kerberos.service_name: HTTP/my-service@realm
 
-  # Name of the Kerberos user. It is used when auth_type is set to password.
+  # Name of the Kerberos user.
   #kerberos.username: elastic
 
   # Password of the Kerberos user. It is used when auth_type is set to password.

--- a/libbeat/outputs/kafka/config_test.go
+++ b/libbeat/outputs/kafka/config_test.go
@@ -38,6 +38,7 @@ func TestConfigAcceptValid(t *testing.T) {
 		"Kerberos with keytab": common.MapStr{
 			"kerberos": common.MapStr{
 				"auth_type":    "keytab",
+				"username":     "elastic",
 				"keytab":       "/etc/krb5kcd/kafka.keytab",
 				"config_path":  "/etc/path/config",
 				"service_name": "HTTP/elastic@ELASTIC",

--- a/metricbeat/metricbeat.reference.yml
+++ b/metricbeat/metricbeat.reference.yml
@@ -1532,7 +1532,7 @@ output.elasticsearch:
   # The service principal name.
   #kerberos.service_name: HTTP/my-service@realm
 
-  # Name of the Kerberos user. It is used when auth_type is set to password.
+  # Name of the Kerberos user.
   #kerberos.username: elastic
 
   # Password of the Kerberos user. It is used when auth_type is set to password.

--- a/packetbeat/packetbeat.reference.yml
+++ b/packetbeat/packetbeat.reference.yml
@@ -1223,7 +1223,7 @@ output.elasticsearch:
   # The service principal name.
   #kerberos.service_name: HTTP/my-service@realm
 
-  # Name of the Kerberos user. It is used when auth_type is set to password.
+  # Name of the Kerberos user.
   #kerberos.username: elastic
 
   # Password of the Kerberos user. It is used when auth_type is set to password.

--- a/winlogbeat/winlogbeat.reference.yml
+++ b/winlogbeat/winlogbeat.reference.yml
@@ -719,7 +719,7 @@ output.elasticsearch:
   # The service principal name.
   #kerberos.service_name: HTTP/my-service@realm
 
-  # Name of the Kerberos user. It is used when auth_type is set to password.
+  # Name of the Kerberos user.
   #kerberos.username: elastic
 
   # Password of the Kerberos user. It is used when auth_type is set to password.

--- a/x-pack/auditbeat/auditbeat.reference.yml
+++ b/x-pack/auditbeat/auditbeat.reference.yml
@@ -852,7 +852,7 @@ output.elasticsearch:
   # The service principal name.
   #kerberos.service_name: HTTP/my-service@realm
 
-  # Name of the Kerberos user. It is used when auth_type is set to password.
+  # Name of the Kerberos user.
   #kerberos.username: elastic
 
   # Password of the Kerberos user. It is used when auth_type is set to password.

--- a/x-pack/filebeat/filebeat.reference.yml
+++ b/x-pack/filebeat/filebeat.reference.yml
@@ -2203,7 +2203,7 @@ output.elasticsearch:
   # The service principal name.
   #kerberos.service_name: HTTP/my-service@realm
 
-  # Name of the Kerberos user. It is used when auth_type is set to password.
+  # Name of the Kerberos user.
   #kerberos.username: elastic
 
   # Password of the Kerberos user. It is used when auth_type is set to password.

--- a/x-pack/metricbeat/metricbeat.reference.yml
+++ b/x-pack/metricbeat/metricbeat.reference.yml
@@ -1940,7 +1940,7 @@ output.elasticsearch:
   # The service principal name.
   #kerberos.service_name: HTTP/my-service@realm
 
-  # Name of the Kerberos user. It is used when auth_type is set to password.
+  # Name of the Kerberos user.
   #kerberos.username: elastic
 
   # Password of the Kerberos user. It is used when auth_type is set to password.

--- a/x-pack/winlogbeat/winlogbeat.reference.yml
+++ b/x-pack/winlogbeat/winlogbeat.reference.yml
@@ -722,7 +722,7 @@ output.elasticsearch:
   # The service principal name.
   #kerberos.service_name: HTTP/my-service@realm
 
-  # Name of the Kerberos user. It is used when auth_type is set to password.
+  # Name of the Kerberos user.
   #kerberos.username: elastic
 
   # Password of the Kerberos user. It is used when auth_type is set to password.

